drivers/wifi/nrf700x/zephyr/src/qspi/src/ficr_prog.c

```diff
@@ -224,64 +225,74 @@ int write_otp_memory(unsigned int otp_addr, unsigned int *write_val)
 		break;
 	case QSPI_KEY:
 		mask_val = QSPI_KEY_FLAG_MASK;
-		write_otp_location(QSPI_KEY, write_val[0]);
-		write_otp_location(QSPI_KEY+1, write_val[1]);
-		write_otp_location(QSPI_KEY+2, write_val[2]);
-		write_otp_location(QSPI_KEY+3, write_val[3]);
+		for (int i = 0; i < QSPI_KEY_LENGTH_BYTES / 4; i++) {
+			ret = write_otp_location(QSPI_KEY + i, write_val[i]);
+			if (ret == -ENOEXEC) {
+				LOG_ERR("Write Exception\n");
```

Comment:
This has to prefix with proper context, else we don't where this print came from.

Reply:
Prefixed the source of failure while printing exception.